### PR TITLE
Signature verification using rc_crypto

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by smarnach** [comment]
@leplatrem Sorry, won't have time for this in the current week. Do you have anyone else who could review this?

**Comment by leplatrem** [comment]
@smarnach It's not really urgent, don't worry :) We can discuss that when you find a moment, or I'll look for someone else next week

File: src/client/signatures/rc_crypto_verifier.rs [hunk 1]
```diff
@@ -16,70 +14,41 @@ impl Verification for RcCryptoVerifier {
...
+            RcErrorKind::CertificateSubjectError => {
+                SignatureError::InvalidCertificateSubject("".to_string())
+            }
+            RcErrorKind::CertificateIssuerError => {
+                SignatureError::InvalidCertificateIssuer("".to_string())
+            }
+            RcErrorKind::CertificateValidityError => SignatureError::CertificateExpired,
+            RcErrorKind::CertificateChainError(_) => SignatureError::CertificateTrustError,
+            _ => SignatureError::MismatchError(err.to_string()),
+        })
```
Comment (mythmon) [inline src/client/signatures/rc_crypto_verifier.rs:52]: Personally, I'd do this as a `impl From<RcErrorKind> for SignatureError`. The code would be pretty much the same, just in a different place. This function would then end up more like 

```rust
contentsignature::verify(...)?;
Ok(())
```

File: src/client/signatures.rs [hunk 2]
```diff
@@ -159,13 +153,13 @@ pub enum SignatureError {
     #[error("certificate could not be downloaded from {}: HTTP {}", response.url, response.status)]
     CertificateDownloadError { response: Response },
     #[error("certificate content could not be parsed: {0}")]
-    CertificateContentError(#[from] x509::X509Error),
+    CertificateContentError(String),
     #[error("root certificate fingerprint has bad format: {0}")]
-    RootFormatError(String),
+    RootHashFormatError(String),
```
Comment (mythmon) [inline src/client/signatures.rs:158]: For what it's worth, I've had a lot of success combining *both* `thiserror` and `anyhow`. The strings here aren't intended to be machine-readable. Using `anyhow` lets you preserver more context while keeping the free form nature that strings provide.

Probably not worth changing in this PR, but maybe something to consider for 1.0?

**Review by mythmon — APPROVED** [review]

File: src/client/signatures.rs (hunk 2 above)
Comment (leplatrem) [reply src/client/signatures.rs:158]: Thanks! I created https://github.com/mozilla-services/remote-settings-client/issues/143

**Pull request merged**